src/Data/Text/AhoCorasick/Searcher.hs

```diff
@@ -152,5 +152,24 @@ containsAny !searcher !text =
     -- On the first match, return True immediately.
     f _acc _match = Aho.Done True
   in case caseSensitivity searcher of
-    CaseSensitive  -> Aho.runText False f (automaton searcher) text
-    IgnoreCase      -> Aho.runLower False f (automaton searcher) text
+    CaseSensitive -> Aho.runText False f (automaton searcher) text
+    IgnoreCase   -> Aho.runLower False f (automaton searcher) text
+
+-- | Returns whether the haystack contains all of the needles.
+-- This function does not use the 'Searcher' type as it has to construct a very specific automaton.
+containsAll :: CaseSensitivity -> [Text] -> Text -> Bool
+containsAll !case_ !ns !haystack =
+  let
+    needleIds = [0..length ns - 1] :: [Int]
+    initial = IS.fromDistinctAscList needleIds
+    ac = Aho.build $ zip (map Utf16.unpackUtf16 ns) needleIds
```

Comment:
Building an automaton for every call is bound to be quite inefficient. A big advantage of Aho-Corasick is that you can use the same automaton many times for different inputs.

My suggestion would be to split this up into a `build`-function for creating a searcher with indexes as values, and have the `containsAll` function take such a searcher.